clippy_lints/src/lib.rs

```diff
@@ -908,6 +909,7 @@ pub fn register_plugins(store: &mut rustc_lint::LintStore, sess: &Session, conf:
     store.register_late_pass(|_| Box::new(fn_null_check::FnNullCheck));
     store.register_late_pass(|_| Box::new(permissions_set_readonly_false::PermissionsSetReadonlyFalse));
     store.register_late_pass(|_| Box::new(size_of_ref::SizeOfRef));
+    store.register_late_pass(|_| Box::new(unnecessary_box_returns::UnnecessaryBoxReturns));
```

Comment:
This lint suggests changing the signature of functions. Therefore, it should take the config value `avoid-breaking-exported-api` into account.

Reply:
I believe I fixed this just now. It's a bit weird using `check_item` now to check functions, but I wasn't able to figure out how to do it with `check_fn`. Let me know if I'm missing something.